integration/testdata/database/DbTest.kt

```diff
@@ -0,0 +1,35 @@
+package ftl.dbtest
+
+import xyz.block.ftl.Context
+import xyz.block.ftl.Verb
+import xyz.block.ftl.Database
+
+data class DbRequest(val data: String?)
+data class DbResponse(val message: String? = "ok")
+
+val db = Database("testdb")
+
+class DbTest {
+  @Verb
+  fun create(context: Context, req: DbRequest): DbResponse {
+    persistRequest(req)
+    return DbResponse()
+  }
+
+  fun persistRequest(req: DbRequest) {
+    db.conn {
```

Comment:
Oh, nice!

Reply:
it turns out `java.sql.Connection` extends `AutoCloseable`, which handles resource management with the caveat that the `close()` logic is only properly utilized if you use this [`.use` function](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/use.html)

so now calling db.conn { ... } creates a new connection and executes the provided block in a `.use` function to make sure the resource is properly managed. also, not sure if this fixed the integration test but with a few other fixes  it's working now!